sway-core/src/semantic_analysis/ast_node/impl_trait.rs

```diff
@@ -329,10 +329,20 @@ fn type_check_trait_implementation(
     // this name space is temporary! It is used only so that the below methods
     // can reference functions from the interface
     let local_namespace: NamespaceRef = create_new_scope(namespace);
+
+    // A trait impl needs access to everything that the trait methods have access to, which is
+    // basically everything in the path where the trait is declared.
+    // First, get the path to where the trait is declared. This is a combination of the path stored
+    // in the symbols map and the path stored in the CallPath.
+    let path_to_symbol = local_namespace.get_path_for_symbol(&trait_name.suffix);
+    let mut real_path = trait_name.prefixes.clone();
+    real_path.extend(path_to_symbol);
```

Comment:
Is there a better way to do this?

Reply:
this seems reasonable, if you really wanted to code golf, you could do something using `.concat()` to make a one-liner